src/main/java/com/firebolt/jdbc/client/FireboltClient.java

```diff
@@ -194,6 +203,49 @@ private String extractErrorMessage(Response response, boolean isCompress) throws
 		return new String(entityBytes, StandardCharsets.UTF_8);
 	}
 
+	private ServerError parseServerError(String responseText) {
+		try {
+			if (responseText == null) {
+				return new ServerError(null, null);
+			}
+			ServerError serverError = new ServerError(new JSONObject(responseText));
+			ServerError.Error[] errors = serverError.getErrors() == null ? null : Arrays.stream(serverError.getErrors()).map(this::updateError).toArray(ServerError.Error[]::new);
+			return new ServerError(serverError.getQuery(), errors);
+		} catch (JSONException e) {
+			String message = responseText;
+			Location location = null;
+			Entry<Location, String> locationAndText = getLocationFromMessage(responseText);
+			if (locationAndText != null) {
+				location = locationAndText.getKey();
+				message = locationAndText.getValue();
+			}
+			return new ServerError(null, new ServerError.Error[] {new ServerError.Error(null, message, null, null, null, null, null, location)});
+		}
+	}
+
+	private ServerError.Error updateError(ServerError.Error error) {
+		if (error == null || error.getDescription() == null) {
+			return error;
+		}
+		Entry<Location, String> locationAndText = getLocationFromMessage(error.getDescription());
```

Comment:
I see that we call `updateError` in case where the error is in a JSON format. Yet I see that we still take the location from the description, while I believe we should take one from the actual `location` json field. Am I missing something?

Reply:
Yes, the case is that right now the location does not present in returned JSON but it can be found in the description. This was the reason to implement this logic. If location is NOT found into description the error is returned as-is (i.e. including location). Otherwise location extracted from the description is used.

Reply:
Here is the example of JSON that I got from the server

```
{
  "errors": [
    {
      "description": "Line 1, Column 8: Column 'wrong' does not exist."
    }
  ],
  "query": {
    "query_id": "77f45647-305e-4611-9537-81bc081dc3f6",
    "request_id": "c263509e-b6b2-49db-8c9b-8dcf274b68c9"
  },
  "statistics": {
    "elapsed": 0.0
  }
}
```

Reply:
According to the design doc, I see that location can be returned as a part of json in `location` field. Do we check if that is present?
```
{
      "code": "CMP0001",
      "name": "<error_name>",
      "severity": "ERROR",
      "source": "User Error",
      "description": "Division by zero",
      "resolution": "",
      "helpLink": "",
      "location": {
        "failingLine": 42,
        "startOffset": 120,
        "endOffset": 135
      }
    }
    ```

Reply:
Exactly. This does not present so far in my tests. I posted here the real JSON that I got from the server. IMHO my implementation supports both cases.

Reply:
Anyway, I have fixed thew logic: if Location exists its value will be used. Only if Location is null but column and line can be extracted from the message we use this information.